apps/hrbfc/hrbfc.star

```diff
@@ -0,0 +1,383 @@
+"""
+Applet: HRBFC
+Summary: H&R Beavers' Live Scores
+Description: Live scores and upcoming match details for the Hampton and Richmond Borough Football Club. Go Beavers!
+Author: Jonathan Damico
+"""
+
+load("cache.star", "cache")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("math.star", "math")
+load("render.star", "render")
+load("schema.star", "schema")
+load("time.star", "time")
+
+DEFAULT_LOCATION = """
+{
+	"lat": "51.41514071311692,",
+	"lng": "-0.3631541244389484",
+	"description": "Hampton & Richmond Borough Football Club Ground (Beveree Stadium)",
+	"locality": "Hampton",
+	"place_id": "ChIJq4On8EULdkgRYjjDaPfI-SQ",
+	"timezone": "Europe/London"
+}
+"""
+
+HRBFC_TEAM_ID = "4680"
+
+BASE_API_URL = "https://26vyvv1b53.execute-api.us-east-1.amazonaws.com"
+GAME_DATA_PATH = BASE_API_URL + "/getCurrentOrNextGame"
+TEAM_NAMES_PATH = BASE_API_URL + "/getTeamNames"
+
+BACKGROUND_IMAGE_URL = "https://hrbfc-png.s3.amazonaws.com/hrbfc-new-darken.png"
+
+FONT = "tb-8"
+
+GAME_STATUSES = {
+    "TBD": ("TBD", "Time To Be Defined"),
+    "NS": ("Not Started", "Not Started"),
+    "1H": ("First Half", "First Half, Kick Off"),
+    "HT": ("Halftime", "Halftime"),
+    "2H": ("Second Half", "Second Half, 2nd Half Started"),
+    "ET": ("Extra Time", "Extra Time"),
+    "BT": ("Break Time", "Break Time"),
+    "P": ("Penalty", "Penalty In Progress"),
+    "SUSP": ("Suspended", "Match Suspended"),
+    "INT": ("Interrupted", "Match Interrupted"),
+    "FT": ("Full Time", "Match Finished"),
+    "AET": ("Full Time", "Match Finished After Extra Time"),
+    "PEN": ("Full Time", "Match Finished After Penalty"),
+    "PST": ("Postponed", "Match Postponed"),
+    "CANC": ("Cancelled", "Match Cancelled"),
+    "ABD": ("Abandoned", "Match Abandoned"),
+    "AWD": ("Tech Loss", "Technical Loss"),
+    "WO": ("WalkOver", "WalkOver"),
+    "LIVE": ("In Progress", "In Progress"),
+}
+
+def countdown_string(timeIn):
+    duration = timeIn - time.now()
+
+    days = math.floor(duration.hours / 24)
+    hours = math.floor(duration.hours - days * 24)
+    minutes = math.floor(duration.minutes - (days * 24 * 60 + hours * 60))
+    if (days > 0):
+        return str(days) + "d " + str(hours) + "h"
+    elif (hours > 0):
+        return str(hours) + "h " + str(minutes) + "m"
+    else:
+        return str(minutes) + "m"
+
+def get_current_or_next_game_data():
+    cached_data = cache.get("hrbfc-data")
```

Comment:
Note that you can pass `ttl_seconds` straight into `http.get()`. This will make the http module handle caching for you. Simplifies the code a bit, and also future proofs against possible upcoming changes to the scoping of the cache module.

Reply:
Done